### ISSUE TITLE
Proper usage of learning rate schedulers, my scheduler doesn't call .step()

### ISSUE THREAD

**Issue by jiwidi** [PyTorchLightning/pytorch-lightning#3869]
#### What is your question?

Hi! 

Is there someone here ho has used learning rate schedulers? I’m using mine following the docs at https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.callbacks.lr_monitor.html#pytorch_lightning.callbacks.lr_monitor.LearningRateMonitor but if I log the learning rate is constant and doesnt use the scheduler. Maybe there is something extra I should add to the code?

Right now im defining it like this:
#### Code

```python
def configure_optimizers(self):
        optimizer = optim.AdamW(self.parameters(), lr=self.hparams.learning_rate)
        lr_scheduler = {'scheduler': optim.lr_scheduler.OneCycleLR(
                                        optimizer,
                                        max_lr=self.hparams.learning_rate,                                                       steps_per_epoch=int(len(self.train_dataloader())),
                                        epochs=self.hparams.epochs,
                                        anneal_strategy="linear",
                                    ),
                        'name': 'learning_rate'}
        return [optimizer], [lr_scheduler]
```
My logs for ‘learning_rate’ are a constant value, the fact that they appear with the name ‘learning_rate’ instead of 'lr_adamW' like the use to tells me that the trainer has catch up the lr_scheduler but is just not calling the step method every mini batch(?) Also, the value reported is the same as the scheduler should be expected to start with.

 - OS: [linux
 - Packaging [conda]
 - Version [0.9.0]

**Comment by williamFalcon** [comment]
solved on slack

**Comment by zbloss** [comment]
What was the issue? I'm experiencing the same problem

**Comment by jiwidi** [comment]
> What was the issue? I'm experiencing the same problem

I had to add the interval key to make it update every step
```python
    def configure_optimizers(self):
        """
        Return whatever optimizers and learning rate schedulers you want here.
        At least one optimizer is required.
        """
        optimizer = optim.AdamW(self.parameters(), lr=self.learning_rate)
        lr_scheduler = {'scheduler': optim.lr_scheduler.OneCycleLR(
                                        optimizer,
                                        max_lr=self.learning_rate,
                                        steps_per_epoch=int(len(self.train_dataloader())),
                                        epochs=self.hparams.epochs,
                                        anneal_strategy="linear",
                                        final_div_factor = 30,
                                    ),
                        'name': 'learning_rate',
                        'interval':'step',
                        'frequency': 1}

        return [optimizer], [lr_scheduler]
```

**Comment by Justin1904** [comment]
Running into the same issue and could solve it by adding interval and frequency key. However, could you share more insights about why you had to specify interval/frequency key to make it work? Reading from the documentation or examples of Lightning it seems to rather indicate that those keys are not mandatory.

**Comment by ilianherzi** [comment]
For version 0.9.0 I'm even with the added keys the learning rate continues to remain constant. Not quite sure if anyone else is experiencing this.

**Comment by mrsndmn** [comment]
> even with the added keys the learning rate continues to remain constant

I had the same issue recently because of this:
```python
self.automatic_optimization = False
```
